src/frontend/citizen-portal/src/app/modules/survey-js/survey-page/survey-page.component.ts

```diff
@@ -1,8 +1,32 @@
-import { Component } from '@angular/core';
+import { Component, OnInit } from '@angular/core';
+import { Router } from '@angular/router';
+import { KeycloakService } from 'keycloak-angular';
 
 @Component({
   selector: 'app-survey-page',
   templateUrl: './survey-page.component.html',
   styleUrls: ['./survey-page.component.scss'],
 })
-export class SurveyPageComponent {}
+export class SurveyPageComponent implements OnInit {
+  userName: string | null;
+
+  constructor(
+    protected keycloakAngular: KeycloakService,
+    protected router: Router
+  ) {}
+
+  public async ngOnInit() {
+    try {
+      const userProfile = await this.keycloakAngular.loadUserProfile();
+      this.userName = `${userProfile.firstName} ${userProfile.lastName}`;
+    } catch (e) {
+      console.log('Failed to load user details', e);
```

Comment:
does it make sense to have a log component?

Reply:
yes, we should